### PR TITLE
AST-238: Add global script

### PR DESCRIPTION
**Pull request by pierallard**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,9 +1,6 @@
-###> symfony/framework-bundle ###
 APP_ENV=dev
 APP_SECRET=b2c78bf3e68a80912087a764c89bf7c7
-###< symfony/framework-bundle ###
-
-AKENEO_API_BASE_URI=
+AKENEO_API_BASE_URI=http://httpd:80
```
Comment (SamirBoulil) [inline .env]: Should we let it as is ?

Comment (pierallard) [reply .env]: We should do a better README instead, yes.

**Review by grena — APPROVED** [review]

File: .env (hunk 1 above)
Comment (SamirBoulil) [reply .env]: YES

File: docker-compose.yml [hunk 2]
```diff
@@ -10,6 +10,7 @@ services:
     volumes:
       - './:/srv/csv-to-asset'
       - '${HOST_COMPOSER_HOME:-~/.composer}:/var/www/.composer'
+      - '/home/pierallard/akeneo/pim-enterprise-dev/:/srv/ee'
```
Comment (SamirBoulil) [inline docker-compose.yml]: revert ?

File: .env (hunk 1 above)
Comment (ValentinMumble) [reply .env]: YES

File: README.md [hunk 3]
```diff
@@ -10,61 +11,41 @@ composer install
...
-
-- For attribute without value per channel/locale:
-    - `<attribute_code>`, eg. `description`
-- For attribute with value per channel:
-    - `<attribute_code>-<channel_code>`, eg. `description-ecommerce`
-- For attribute with value per locale
-    - `<attribute_code>-<locale_code>`, eg. `description-en_US`
-- For attribute with value per channel and per locale:
-    - `<attribute_code>-<locale_code>-<channel_code>`, eg. `description-en_US-mobile` (_locale first_)
+## Docker setup [optionall]
```
Comment (SamirBoulil) [inline README.md]:
```suggestion
## Docker setup [optional]
```

File: README.md [hunk 4]
```diff
@@ -10,61 +11,41 @@ composer install
...
 
-So, let's imagine this structure for the `brand` reference entity:
-- A code
-- A description with one value per locale
-- Some tags (an attribute with multiple options)
+- Update your Enterprise Edition docker-compose file to allowe external data from network
```
Comment (SamirBoulil) [inline README.md]:
```suggestion
- Update your Enterprise Edition docker-compose file to allow external data from network
```

File: README.md [hunk 5]
```diff
@@ -10,61 +11,41 @@ composer install
...
-``` 
+Run `./bin/migrate.php <family-code> <path-to-ee-installation>`
 
-So if you want to import your records in the .csv file located in `/tmp/file.csv` for your `brand` reference entity:
-```bash
-php bin/console app:import /tmp/file.csv brand
-```
+Note: Using docker, your `path-to-ee-installation` will be `/srv/ee`.
+
+This command will
```
Comment (SamirBoulil) [inline README.md]:
```suggestion
This command will:
```

File: README.md [hunk 6]
```diff
@@ -10,61 +11,41 @@ composer install
...
+- Export the former Akeneo PAM tables and put it into temporary folder in CSV format
```
Comment (SamirBoulil) [inline README.md]:
```suggestion
- Export the PAM assets in CSV into a temporary folder
```

File: bin/migrate.php [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        die($process->getExitCode());
+    } else {
+        $output = $process->getOutput();
+        $io->write($output);
+        $callback($output);
+    }
+}
+
+executeCommand(
+    ['bin/console', sprintf('--env=%s', $_SERVER['APP_ENV']), 'pimee:migrate-pam-assets:export-assets', '/tmp'],
```
Comment (SamirBoulil) [inline bin/migrate.php]: in 4.0 shouldn't we execute the commands as so:
```
APP_ENV=prod bin/console ........
```
I don't know if it's equivalent to `--env=prod`

Comment (pierallard) [reply bin/migrate.php]: I think I can drop this one to use the default EE env.

File: src/Command/MigrateCommand.php [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->io = new SymfonyStyle($input, $output);
+        $this->assetFamilyCode = $input->getArgument('assetFamilyCode');
+
+        $tmpfname = tempnam('/tmp', 'migration_target_');
+
+        $this->executeCommand('app:create-family', [$this->assetFamilyCode]);
+        $this->executeCommand('app:merge-files', ['/tmp/assets.csv', '/tmp/variations.csv', $tmpfname]);
```
Comment (SamirBoulil) [inline src/Command/MigrateCommand.php]: Can you make that "/tmp/assets.csv" are given as arguments of this commands ?

**Pull request merged**